index.js

```diff
@@ -126,6 +128,89 @@ function invokePromisedTaskFunction(taskFunction, cumulusMessage, context) {
   });
 }
 
+/**
+ * Get granules from execution message.
+ *   Uses the order of precedence as defined by the cumulus/common/message
+ *   description.
+ *
+ * @param {Object} message - An execution message
+ * @returns {Array<Object>} - An array of granule ids
+ */
+function getMessageGranules(message) {
+  const granules = get(message, 'payload.granules')
+    || get(message, 'meta.input_granules')
+    || get(message, 'cma.event.payload.granules')
+    || get(message, 'cma.event.meta.input_granules');
+
+  if (granules) {
+    return granules.map((granule) => granule.granuleId);
+  }
+
+  return [];
+}
+
+/**
+ * Get the stackname pulled from the meta of the event.
+ *
+ * @param {Object} message - A cumulus event message.
+ * @returns {string} - The cumulus stack name.
+ */
+function getStackName(message) {
+  return get(message, 'meta.stack')
+    || get(message, 'cma.event.meta.stack');
+}
+
+/**
+ * Gets parent arn from execution message.
+ *
+ * @param {Object} message - An execution message.
+ * @returns {string} - the parent execution.
+ */
+function getParentArn(message) {
+  return get(message, 'cumulus_meta.parentExecutionArn')
+    || get(message, 'cma.event.cumulus_meta.parentExecutionArn');
+}
+
+/**
+ * Get current execution name from Cumulus message.
+ *
+ * @param {Object} message - Cumulus message.
+ * @returns {string} current execution name.
+ */
+function getExecutions(message) {
+  return get(message, 'cumulus_meta.execution_name')
+    || get(message, 'cma.event.cumulus_meta.execution_name');
+}
+
+
+/**
+ * Conditionally set environment variable when targeted value is not undefined.
+ *
+ * @param {string} VARNAME - environment variable name
+ * @param {string} value - value to set variable to if not undefined
+ * @returns {undefined} - none
+ */
+function safeSetEnv(VARNAME, value) {
+  if (value !== undefined) process.env[VARNAME] = value;
+}
+
+/**
+ * Set environment variables to be used in logging based on the Cumulus event
+ * message and context that runCumulusTask was invoked.
+ *
+ * @param {Object} cumulusMessage - cumulus event message
+ * @param {Object} context - lambda context object.
+ * @returns {undefined} - no return values
+ */
+function setCumulusEnvironment(cumulusMessage, context) {
+  safeSetEnv('EXECUTIONS', getExecutions(cumulusMessage));
+  safeSetEnv('SENDER', context.functionName);
+  safeSetEnv('TASKVERSION', context.functionVersion);
+  safeSetEnv('STACKNAME', getStackName(cumulusMessage));
+  safeSetEnv('GRANULES', JSON.stringify(getMessageGranules(cumulusMessage)));
```

Comment:
Do we have concern here about running into possible max size issues with this environment variable? In a discover granules situation, there could be thousands of granules. 

https://devblogs.microsoft.com/oldnewthing/20100203-00/?p=15083

Reply:
@markdboyd @flamingbear maybe we should limit the logging to 100 granules? What do you think?

Reply:
since we're down to just the granuleId. I think I looked up that we can have 32k characters.  But We probably should limit it somehow, but not sure how.  100 seems small.

Reply:
I guess it's actually 32k characters for all env vars.